rewriter/ClassNew.cc

```diff
@@ -82,4 +70,88 @@ vector<ast::ExpressionPtr> ClassNew::run(core::MutableContext ctx, ast::Assign *
     return stats;
 }
 
+vector<ast::ExpressionPtr> rewriteWithBind(core::MutableContext ctx, ast::Send *send) {
+    vector<ast::ExpressionPtr> empty;
+
+    auto recv = ast::cast_tree<ast::UnresolvedConstantLit>(send->recv);
+    if (recv == nullptr) {
+        return empty;
+    }
+
+    if (!ast::isa_tree<ast::EmptyTree>(recv->scope) || recv->cnst != core::Names::Constants::Class() ||
+        send->fun != core::Names::new_()) {
+        return empty;
+    }
+
+    auto argc = send->numPosArgs();
+    if (argc > 1 || send->hasKwArgs()) {
+        return empty;
+    }
+
+    if (argc == 1 && !ast::isa_tree<ast::UnresolvedConstantLit>(send->getPosArg(0))) {
+        return empty;
+    }
+
+    auto *block = send->block();
+    if (block == nullptr) {
+        return empty;
+    }
+
+    ast::ExpressionPtr type;
+
+    if (argc == 0) {
+        type = ast::MK::Constant(send->loc, core::Symbols::Class());
+    } else {
+        auto target = send->getPosArg(0).deepCopy();
+        type = ast::MK::ClassOf(send->loc, std::move(target));
+    }
+
+    auto bind = ast::MK::Bind(send->loc, ast::MK::Self(send->loc), std::move(type));
+
+    ast::InsSeq::STATS_store blockStats;
+    blockStats.emplace_back(std::move(bind));
+
+    if (auto insSeq = ast::cast_tree<ast::InsSeq>(block->body)) {
+        for (auto &&stat : insSeq->stats) {
```

Comment:
I think using `&&` here is harmless, but it looks like the deduced type will be `ExpressionPtr &` so I would drop the extra `&`. Here's a similar example on godbolt showing the deduced type for an element of a vector of unique_ptrs: https://godbolt.org/z/TW1q64r8s
```suggestion
        for (auto &stat : insSeq->stats) {
```